cpmpy/expressions/globalconstraints.py

```diff
@@ -191,7 +192,7 @@ class AllDifferentExcept0(GlobalConstraint):
     """
     def __init__(self, *args):
         flatargs = flatlist(args)
-        if not (all(is_boolexpr(arg) for arg in flatargs) or not any(is_boolexpr(arg) for arg in flatargs)):
+        if not is_single_type_list(flatargs):
```

Comment:
I am not convinced we need any of these 'single_type_list' checks...

If you can use booleans as integers, then any of them should be valid with a mix of bool/int too... this is not a typecheck restriction. If a solver does not support it, we need to transform it later for that solver

Reply:
we are going from 'no type checking' to 'very strict' type checking... I would say to drop the 'is_single_type_list' checks (do keep the IfThenElse one)

Reply:
What about checking if we get lists where we expect lists and vice versa? did it for Count now, but might be overkill